panoptes_cli/commands/subject_set.py

```diff
@@ -11,15 +12,91 @@
 
 from pathvalidate import is_valid_filename, sanitize_filename
 
+from PIL import Image
+
 from panoptes_cli.scripts.panoptes import cli
 from panoptes_client import SubjectSet
 from panoptes_client.panoptes import PanoptesAPIException
+from panoptes_client.subject import UnknownMediaException
+import logging
+
+try:
+    import magic
+    MEDIA_TYPE_DETECTION = 'magic'
+except ImportError:
+    import pkg_resources
+    try:
+        pkg_resources.require("python-magic")
+        logging.getLogger('panoptes_client').warn(
+            'Broken libmagic installation detected. The python-magic module is'
+            ' installed but can\'t be imported. Please check that both '
+            'python-magic and the libmagic shared library are installed '
+            'correctly. Uploading media other than images may not work.'
+        )
+    except pkg_resources.DistributionNotFound:
+        pass
+    import imghdr
+    MEDIA_TYPE_DETECTION = 'imghdr'
+
+_OLD_STR_TYPES = (str,)
+try:
+    _OLD_STR_TYPES = _OLD_STR_TYPES + (unicode,)
+except NameError:
+    pass
 
+IMG_MIME_TYPES = ['image/jpeg']
 LINK_BATCH_SIZE = 10
 MAX_PENDING_SUBJECTS = 50
 MAX_UPLOAD_FILE_SIZE = 1024 * 1024
 CURRENT_STATE_VERSION = 1
 
+
+def compress_image(image_path, save_quality=70):
+    """ Resize and compress a single image and return Byte object """
+    # Read the file from disk
+    mime_type = get_mime_type(image_path)
+
+    if mime_type not in IMG_MIME_TYPES:
+        raise UnknownMediaException(
+            f"Media type {mime_type} is not an image format"
```

Comment:
More appropriate to say "not JPEG", as PNG is image format but not one for which compression is currently supported.